UI/Selenium/Steps/SelectHearingSteps.cs

```diff
@@ -44,10 +48,12 @@ public void ThenHearingIdShouldBeCopied()
         [When(@"I click on link to join by Quicklink details to clipboard")]
         public void WhenIClickOnLinkToJoinByQuicklinkDetailsToClipboard()
         {
-            ExtensionMethods.MoveToElement(Driver, SelectHearingPage.Quicklinks, _scenarioContext);
-            ExtensionMethods.FindElementWithWait(Driver, SelectHearingPage.QuicklinkCopy, _scenarioContext).Click();
-            ApplicationData.hearingListUrl = new TextCopy.Clipboard().GetText();
-
+            if (ExtensionMethods.WaitForPageLoad(Driver, SelectHearingPage.Quicklinks, _scenarioContext))
+            {
+                ExtensionMethods.MoveToElement(Driver, SelectHearingPage.Quicklinks, _scenarioContext);
+                ExtensionMethods.FindElementWithWait(Driver, SelectHearingPage.QuicklinkCopy, _scenarioContext).Click();
+                ApplicationData.hearingListUrl = new TextCopy.Clipboard().GetText();
```

Comment:
we're using ScenarioContext to share data between steps. I think we should use same approach here instead of ApplicationData

Reply:
changed